app/authentication/user_id_generator.py

```diff
@@ -0,0 +1,30 @@
+from app import settings
+from app.metadata.metadata_store import MetaDataConstants
+from app.utilities.strings import to_str
+import hashlib
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class UserIDGenerator(object):
+
+    @staticmethod
+    def generate_id(token):
```

Comment:
I did consider that, I think its the Java in me that wants to put everything into classes.